draft-ietf-mls-protocol.md

```diff
@@ -1686,24 +1626,70 @@ decryption.
 +-+-+-+-+---------...---+
 ~~~~~
 
-The Additional Authenticated Data (AAD) input to the encryption
-contains an object of the following form, with the values used to
-identify the key and nonce:
+## Masking Sender Data
```

Comment:
Since we're going to have an auth tag anyway, it seems like we could simplify this a fair bit by just using the ciphersuite's AEAD to encrypt the SenderData (as we do now), and just sampling the *nonce* from the ciphertext.

That obviously removes the need to do XOR in addition to normal AEAD, and may allow us to simplify the sampling as well.  Since the nonce only needs to be random and not secret, we might be able to get away with just using the first `nonce_size` bytes of the ciphertext (or HKDF-Extract thereof).  So it gets rid of the AES-ECB dependency as well.

And since this changes from (AEAD + XOR + AEAD) to (AEAD + AEAD), we no longer need the `authentication_key`.

```
ciphertext = AEAD.Seal(key=generation.key,
                       nonce=generation.nonce,
                       pt=content,
                       aad=header_with_plaintext_sender_data)
sender_data_nonce = ciphertext[:nonce_size]
encrypted_sender_data = AEAD.Seal(key=sender_data_key,
                                  nonce=sender_data_nonce,
                                  pt=sender_data,
                                  aad=header_without_sender_data_and_with_ciphertext)
```